src/types_.py

```diff
@@ -444,6 +564,34 @@ class IndexContentsListItem(typing.NamedTuple):
     rank: int
     hidden: bool
 
+    @property
+    def table_path(self) -> TablePath:
+        """The table path for the item.
+
+        In the case of a HTTP reference, changes http://canonical.com/1 to http,canonical,com,1
+        removing the HTTP protocol characters so that the path conforms to the path in the non-HTTP
+        case. Any remaining characters not allowed in the path are also removed. For a non-HTTP
+        case, removes the file suffix and splits on / to built the path.
+
+        Returns:
+            The table path for the item.
+        """
+        if self.reference_value.lower().startswith("http"):
+            transformed_reference_value = (
+                self.reference_value.replace("//", "/")
+                .replace(".", "/")
+                .replace("?", "/")
+                .replace("#", "/")
+            )
```

Comment:
It is small stuff, but why not using regex then? it seems also more consistent with the next line

Reply:
The string replace is easier to read, the regex replace is intended to capture any other characters that are not allowed on the `path` column - it would be nicer to do that as a string replace its just that the definition of what is allowed on the path column is already defined as a regex